scheduler/stack.go

```diff
@@ -35,10 +35,12 @@ type Stack interface {
 }
 
 type SelectOptions struct {
-	PenaltyNodeIDs map[string]struct{}
-	PreferredNodes []*structs.Node
-	Preempt        bool
-	AllocName      string
+	PenaltyNodeIDs          map[string]struct{}
+	PreferredNodes          []*structs.Node
+	Preempt                 bool
+	AllocName               string
+	AllocID                 string
```

Comment:
I don't think we're using the alloc ID anymore, are we?

Reply:
ah that's some leftovers from the refactor, i'll clean up